test/functional/test_framework/test_node.py

```diff
@@ -141,12 +139,22 @@ def node_encrypt_wallet(self, passphrase):
         self.rpc = None
         self.rpc_connected = False
 
+    def cli(self, **kwargs):
+        """Creates a configurable cli for the node.
+
+        If arguments is empty then -datadir is set to match the node
+        """
+        args = ["-" + str(key) + ("=" + str(value) if value else "") for (key, value) in kwargs.items()]
+        if not args:
```

Comment:
I think you should add an error here. If args is specified, then `datadir` must be one of the args (otherwise `bitcoin-cli` will look in `~/.bitcoin`).

Reply:
Another option would be to set `datadir=self.datadir` if it isn't overridden otherwise. As you say, there is no reason the argument should ever be *missing*.

Reply:
I think either approach is fine. Making it an error to not include it in args makes it more explicit to the test writer (ie datadir doesn't get set magically for them).